### PR TITLE
Add Django Rest Framework

### LINKED ISSUE #23: Add Django Rest Framework to the project
[Django REST Framework](http://www.django-rest-framework.org/) is a Python/Django library that allows us to create APIs.

Interesting reads:

* [The tutorial](http://www.django-rest-framework.org/tutorial/quickstart/)
* [API Documentation](http://www.django-rest-framework.org/topics/documenting-your-api/)
* [Everything under API guides, depending on the occasion](http://www.django-rest-framework.org/)

The first thing would be to:

* Add basic settings
* Create a [view](http://www.django-rest-framework.org/api-guide/views/) that when hit, will return the following JSON

```
[
    {
        "kind": "email|email_host|ip",
        "value": ":the-value",
        "result": true|false
    },
    ...
]
```

The view should replace the current blacklist view.

### PR DESCRIPTION
**Pull request by nzisouli**
Add Django Rest Framework to the project and create a view that when hit, will return a JSON response. Closes #23

### CODE REVIEW

**Comment by nzisouli** [comment]
Finish work

File: bouncer/blacklist/tests.py [hunk 1]
```diff
@@ -1,6 +1,7 @@
 from django.test import TestCase
 from blacklist import models
 from django.db.utils import IntegrityError
+from rest_framework.response import Response
```
Comment (akalipetis) [inline bouncer/blacklist/tests.py]: I think this is not needed here.

File: bouncer/blacklist/urls.py [hunk 2]
```diff
@@ -1,6 +1,6 @@
-from django.urls import path
-
-from . import views
+from django.urls import path, include
```
Comment (akalipetis) [inline bouncer/blacklist/urls.py]: `include` is not used, please remove it.

File: bouncer/blacklist/urls.py [hunk 3]
```diff
@@ -1,6 +1,6 @@
...
+from rest_framework import routers
```
Comment (akalipetis) [inline bouncer/blacklist/urls.py]: `routers` are not used, please remove it.

File: bouncer/blacklist/views.py [hunk 4]
```diff
@@ -11,32 +8,33 @@
...
-class RequestView(View):
-    template = loader.get_template("blacklist/request.html")
-
-    def get(self, request, *args, **kwargs):
-        context = {"results": []}
-        Row = namedtuple("Row", ["kind", "value", "result"])
-        email_query = self.request.GET.getlist("email")
+class RequestView(views.APIView):
+    def get(self, request):
+        print("GET METHOD")
```
Comment (akalipetis) [inline bouncer/blacklist/views.py]: Please remove `print` statement.

File: bouncer/blacklist/views.py [hunk 5]
```diff
@@ -11,32 +8,33 @@
...
+        result_list = []
+        email_query = request.GET.getlist("email")
         for email in email_query:
-            result = "NO"
+            result = False
```
Comment (akalipetis) [inline bouncer/blacklist/views.py]: This could just be `result = is_email...`

File: bouncer/blacklist/views.py [hunk 6]
```diff
@@ -11,32 +8,33 @@
...
-                result = "YES"
-            row = Row("ip", ip, result)
-            context["results"].append(row)
+                result = True
+            response = {"kind": "ip", "value": ip, "result": result}
+            result_list.append(response)
+        return Response(result_list)
 
-        return HttpResponse(self.template.render(context, request))
+    @classmethod
```
Comment (akalipetis) [inline bouncer/blacklist/views.py]: Is this needed here?

**Review by akalipetis — CHANGES_REQUESTED** [review]
Nice start! 👍 

I have added some first comments, also you need to lock your dependencies again.

Please add `pipenv install --dev --system --deploy` to Travis to avoid such issues in the future.

File: bouncer/blacklist/views.py [hunk 7]
```diff
@@ -10,31 +10,23 @@
...
-            result = False
-            if is_ip_blacklisted(ip):
-                result = True
+            result = is_ip_blacklisted(ip)
             response = {"kind": "ip", "value": ip, "result": result}
             result_list.append(response)
         return Response(result_list)
 
-    @classmethod
     def get_extra_actions(cls):
```
Comment (akalipetis) [inline bouncer/blacklist/views.py]: Why do you need to define this method?

**Review by akalipetis — CHANGES_REQUESTED** [review]
One more minor change.

**Review by akalipetis — APPROVED** [review]
LGTM 🐻

**Pull request merged**